### PR TITLE
Gracefully shutdown opendevin

### PR DESCRIPTION
**Pull request by li-boxuan**
This PR does the following:
- Log shutdown events
- Remove signal handlers that mess up shutdown

### CODE REVIEW

**Comment by li-boxuan** [comment]
Prior to this PR:

image

The full error messages are too long and exceed one screen.

With this PR:

image

You might wonder why we return exit-code 1. It took me a long time to find out that this is by design of `poetry run`.

File: opendevin/server/agent/manager.py [hunk 1]
```diff
@@ -34,11 +32,7 @@ async def dispatch(self, sid: str, action: str | None, data: dict):
 
         await self.sid_to_agent[sid].dispatch(action, data)
 
-    def handle_signal(self, signum, _):
-        print(f'Received signal {signum}, exiting...')
-        self.close()
```
Comment (li-boxuan) [inline opendevin/server/agent/manager.py:39]: This is not important anymore, but just FYI: this is a bug. Upon signal handler being called, `self.close()` is called here. Then at exit, `self.close` is called again by `atexit`.

**Comment by Umpire2018** [comment]
That's great!

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
Thanks! Looks much cleaner

**Pull request merged**